### PR TITLE
chore: delete the configuration related to github cache

### PR DESCRIPTION
**Pull request by tanruixiang**
## Rationale
Due to the huge cache of rust before, unit tests often failed due to insufficient capacity. [CI without cache](https://github.com/CeresDB/ceresdb/actions/runs/6492219953/job/17630769103?pr=1259) takes an acceptable time and does not cause failure problems.

https://github.com/CeresDB/ceresdb/pull/1261 show CI always fail in Rust Cache.
The github action's time seems have huge volatile, https://github.com/CeresDB/ceresdb/actions/runs/6517181839/job/17701414760?pr=1259 takes 24 min without cache, some pr have cache even is slower than this action
## Detailed Changes
Delete out the configuration related to github cache

## Test Plan
CI

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -56,26 +56,8 @@ jobs:
       - run: |
           rustup set auto-self-update disable
           rustup toolchain install ${RUST_VERSION} --profile minimal
-      - name: Release Disk Quota
```
Comment (ShiKaiWi) [inline .github/workflows/ci.yml:59]: The disk may be not enough to run the ci, so how about keeping this step?

Comment (tanruixiang) [reply .github/workflows/ci.yml]: Solved

**Review by jiacai2050 — APPROVED** [review]
LGTM

**Comment by jiacai2050** [comment]
I plan to merge to main and see if it works well in daily dev.

**Pull request merged**